### DISCUSSION TITLE
Configuration based on IMEI

### DISCUSSION THREAD

**Discussion by hellkaim** [IchthysMaranatha/asterisk-chan-quectel#32] [Ideas]
Hi,
The original chan dongle has an option where I can specify a modem IMEI as a reference for dongleX assignment.

`[dongle2]
imei=358711037755485`

If defined in such a manner ttyUSB devices are selected automatically for a given modem.

It is very useful if more then 1 modem is connected to the system and prevents mess with ttyUSB devs after reboot. It well may happen that during initialization ttyUSB devices will have different numbers then on previous boot. Yes, it may be solved with manual udev configuration but hey ))
IMEI identity helps sorting things out easily.

Could you add similar functionality to the code so that I as an admin can define 1 universal sign and the rest will be handled by the chan code? It is ok if the  UAC config should be done manually in case used but we can discuss that.

My personal issue is that I am at no way a programmer and even if I can locate the code responsible for that in the original chan dongle code base I will be unable to change it properly ((

**Comment by IchthysMaranatha** [comment]
I've not tested it, but I don't think Quectel and Simcom modems will support that. Even for chan_dongle it works only with certain Huawei models.

**Comment by hellkaim** [comment]
Ok, so the only way to handle several devices is to somehow hardcore them in udev rules?

**Comment by IchthysMaranatha** [comment]
I think so

**Comment by IchthysMaranatha** [comment]
I've tested out the imei/imsi discovery and got it working. Will update code in some time.

**Reply by hellkaim** [reply]
Hi, pulled a fresh copy, compiled and no - it seems it can't discover modems by IMIE ((

**Reply by hellkaim** [reply]
to be exact:
`[Oct 26 17:01:55] ERROR[2074]: dc_config.c:57 dc_uconfig_fill: Skipping device quectel0. Missing required audio setting
[Oct 26 17:01:55] ERROR[2074]: dc_config.c:57 dc_uconfig_fill: Skipping device quectel1. Missing required audio setting
[Oct 26 17:01:55] ERROR[2074]: dc_config.c:57 dc_uconfig_fill: Skipping device quectel2. Missing required audio setting`

So it seems to find the modem, but my EG25-G does produce 4 devices. I am unable to find the way to disable _wwanX_ devices but even if I can there will be 1 spare left. SO what I did I made a normal config using `data` & `audio` and it seem to work (well calls are not passing through at the moment, but anyway and I think it is die to the power issues).

**Comment by IchthysMaranatha** [comment]
For discovery, check the vendor id and product id and add a row after this if different from ec25 (2c7c, 0125) here https://github.com/IchthysMaranatha/asterisk-chan-quectel/blob/b3534eaedce52819ee3f61db6d191ed35177b955/pdiscovery.c
Your problem seems to be urc as mentioned before also when using serial audio, ttyusb3 is modem port which also accepts at commands but does not receive urc by default. ttyusb2 is the usbat port, try with that. Make sure gps is disabled on the shared audio port.

**Reply by hellkaim** [reply]
First of all thx for prompt reply.
Ok, for whatever reason my git update was unsuccessful. re-synced from scratch and youp - auto discovery works.
Despite my modem was listed in the mentioned array.

**Reply by hellkaim** [reply]
>**Make sure gps is disabled on the shared audio port**

For those who like myself will struggle to remember where and how:
Go _"Quectel_EC25EC21_GNSS_AT_Commands_Manual_V1.1.pdf"_ guide and look for `AT+QGPSCFG=?` the way to set GNSS output to not use USB or UART is a command `AT+QGPSCFG="outport"` that should give you `+QGPSCFG: "outport",none`

P.S. There is also an interesting data in [this guide](https://www.embeddedpi.com/documentation/3g-4g-modems/quecetel-modem-gps-setup) might help if others will come here to check.

**Reply by hellkaim** [reply]
P.S.S And for all Quectel EC25G command guides please check [this GitHub collection](https://github.com/OLIMEX/USB-gLINK/find/master)

**Reply by MikasaGitHub** [reply]
hello ,Will asterisk be disconnected if multiple devices call different people at the same time and stay online?

**Reply by hellkaim** [reply]
@MikasaGitHub No, Asterisk will handle multiple calls just fine.
Think of modems as trunks (technically not correct but still).
The main issue is power thus, not Asterisk.

**Reply by hellkaim** [reply]
@MikasaGitHub more over, if your operator/plan/sim allows conferencing that is also possible.

**Reply by MikasaGitHub** [reply]
ok Thank you so much

**Comment by IchthysMaranatha** [comment]
Sorry, correct row link is https://github.com/IchthysMaranatha/asterisk-chan-quectel/blob/b3534eaedce52819ee3f61db6d191ed35177b955/pdiscovery.c#L71
